src/Services/MailSender.php

```diff
@@ -85,6 +101,34 @@ private function getCurrentSalesChannelId(): ?string
         return $salesChannelId;
     }
 
+    private function getSourceMailId(Context $context): ?string
+    {
+        $request = $this->requestStack->getMainRequest();
+        if ($request === null) {
+            return null;
+        }
+
+        $route = $request->attributes->get('_route');
+        if ($route !== 'api.action.frosh-mail-archive.resend-mail') {
+            return null;
+        }
+
+        $sourceMailId = $request->request->get('mailId');
+
+        if (!\is_string($sourceMailId)) {
+            throw MailArchiveException::parameterMissing('mailId in request');
+        }
+
+        /** @var MailArchiveEntity|null $sourceMail */
+        $sourceMail = $this->froshMailArchiveRepository->search(new Criteria([$sourceMailId]), $context)->first();
+        if (!$sourceMail) {
+            return null;
+        }
+
+        // In case the source Mail is a resend, we want to save the original source mail id
+        return $sourceMail->getSourceMailId() ?? $sourceMailId;
```

Comment:
```suggestion

        // In case the source Mail is a resend, we want to save the original source mail id
        return $sourceMail?->getSourceMailId() ?? $sourceMailId;
```